### PR TITLE
[bugfix] Add job options from the command-line to processor auto-detection remote job

### LINKED ISSUE #3218: Job options passed from the cli are not respected by the autodetection remote job

### PR DESCRIPTION
**Pull request by ekouts**
Closes #3218.

### CODE REVIEW

File: reframe/frontend/cli.py [hunk 1]
```diff
@@ -1110,7 +1110,20 @@ def restrict_logging():
 
         sys.exit(0)
 
-    autodetect.detect_topology()
+    # Need to parse the cli options before autodection
```
Comment (vkarak) [inline reframe/frontend/cli.py:1113]:
```suggestion
    # Need to parse the cli options before autodetection
```

File: reframe/frontend/autodetect.py [hunk 2]
```diff
@@ -136,7 +136,8 @@ def _is_part_local(part):
             part.launcher_type.registered_name == 'local')
 
 
-def _remote_detect(part):
+def _remote_detect(part, job_options=None):
+    job_options = [] if job_options is None else job_options
```
Comment (vkarak) [inline reframe/frontend/autodetect.py:140]: I would avoid having another default here. This is an internal function after all:

```suggestion
def _remote_detect(part, job_options):
```

Comment (vkarak) [reply reframe/frontend/autodetect.py:140]: Maybe also call the argument `cli_job_options` to match the name we use in the scheduler backends.

File: reframe/frontend/autodetect.py [hunk 3]
```diff
@@ -280,7 +282,8 @@ def detect_topology():
                 _save_info(topo_file, part.processor.info)
             elif detect_remote_systems:
                 with runtime.temp_environment(modules=modules, env_vars=vars):
-                    part._processor = ProcessorInfo(_remote_detect(part))
+                    part._processor = ProcessorInfo(_remote_detect(
+                                                    part, job_options))
```
Comment (vkarak) [inline reframe/frontend/autodetect.py:286]: Better reformat as follows; it's cleaner:

```suggestion
                    part._processor = ProcessorInfo(
                        _remote_detect(part, job_options)
                    )
```

File: reframe/frontend/autodetect.py (hunk 2 above)
Comment (ekouts) [reply reframe/frontend/autodetect.py]: Ah, I didn't notice the second comment. Do you mean in `_remote_detection` only or in both `_remote_detection` and `detect_topology`?

**Review by vkarak — APPROVED** [review]
Lgtm

**Pull request merged**